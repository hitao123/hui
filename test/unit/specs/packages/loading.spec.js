import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Loading from '../../../../packages/loading';
import LoadingDemo from '../../../../docs/src/views/loading';

describe('Loading component', () => {
  it('should Loading as Circular', () => {
    const warpper = shallowMount(Loading, {
      propsData: {
        type: 'spinner',
        color: 'white'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should Loading as Spinner', () => {
    expect(renderToString(LoadingDemo)).toMatchSnapshot();
  });
});
