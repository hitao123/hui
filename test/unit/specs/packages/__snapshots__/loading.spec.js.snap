// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should Loading as Circular 1`] = `<div class="h-loading h-loading--spinner h-loading--white"><span class="h-loading__spinner h-loading__spinner--spinner"><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i> <!----></span></div>`;

exports[`Loading component should Loading as Spinner 1`] = `
<div>
  <div class="demo-loading">
    <h3>circle</h3>
    <h-loading color="black"></h-loading>
    <h-loading color="white"></h-loading>
  </div>
  <div class="demo-loading">
    <h3>spinning</h3>
    <h-loading type="spinner" color="black"></h-loading>
    <h-loading type="spinner" color="white"></h-loading>
  </div>
</div>
`;
